pkg/asset/agent/mirror/registriesconf.go

```diff
@@ -208,6 +214,20 @@ func (i *RegistriesConf) finish() error {
 	return nil
 }
 
+func (i *RegistriesConf) validateReleaseImageIsSameInRegistriesConf(source string) error {
+
+	releaseImage := &releaseimage.Image{}
+	releaseImage.Generate(asset.Parents{})
+
+	releaseImagePath := strings.Split(releaseImage.PullSpec, ":")[0]
+
+	if !strings.Contains(string(i.File.Data), releaseImagePath) {
+		return errors.New(fmt.Sprintf("%s %s should have an entry matching the releaseImage %s", source, RegistriesConfFilename, releaseImagePath))
```

Comment:
I like the fact that the source of the setting is now identified but I'd suggest some way to point out where the faulty mirror setting was input, as "Generated" doesn't really give a clue. Maybe something like, for example:
```
if source == "Generated {
    return errors.New(fmt.Sprintf("The ImageContentSource configuration in install-config.yaml should have an entry matching the releaseImage %s", releaseImagePath)
} else {
   return errors.New(fmt.Sprintf("%s should have an entry matching the releaseImage %s", RegistriesConfFilename, releaseImagePath)
}
```

Reply:
I had thought about a similar message earlier but later discarded it with my assumption that users know the difference between generated from install-config vs reading from the disk. I am happy to include this suggestion. Thank you.